Sources/Lifecycle/Lifecycle.swift

```diff
@@ -517,7 +520,9 @@ public class ComponentLifecycle: LifecycleTask {
             return callback(index, nil)
         }
         self.log("starting tasks [\(tasks[index].label)]")
+        let startTime = DispatchTime.now()
         start { error in
+            Timer(label: "\(self.label).\(tasks[index].label).lifecycle.start").recordNanoseconds(DispatchTime.now().uptimeNanoseconds - startTime.uptimeNanoseconds)
```

Comment:
Could it be worth making some struct/enum with the labels such that it is easier to look in one place to see "aha, these are all the metric labels this lib exports"?

I do this in some other project, it is easier to know what metrics are exposed then.
Sure not all that much here but it's a nice pattern

Reply:
We have `    func recordInterval(since: DispatchTime, now: DispatchTime = .now()) {` for this type of nanosecond recording, consider using it

Reply:
recordInterval is not define in the core library afaict

Reply:
We have it here https://github.com/apple/swift-metrics/blob/44e8bfc7f5a0686e21d5d21a7ea5454e74ed3449/Sources/Metrics/Metrics.swift#L42-L44

maybe it's not released yet? 🤔

Reply:
I see but it's not in CoreMetrics hm

Reply:
ah yes... I prefer not to introduce dependency on Foundation  for low level libs if we can get away with it

Reply:
Yeah though this could live in core I guess, I'll ticketify -- it's just Dispatch dep, not foundation hm